pymatgen/io/packmol.py

```diff
@@ -77,8 +78,10 @@ def run(self, path: Union[str, Path], timeout=30):
         except subprocess.CalledProcessError as e:
             raise ValueError("Packmol failed with errorcode {} and stderr: {}".format(e.returncode, e.stderr)) from e
         else:
-            with open(self.stdoutfile, "w") as out:
+            with open(Path(path, self.stdoutfile), "w") as out:
                 out.write(p.stdout.decode())
+        finally:
+            os.chdir(wd)
```

Comment:
In the previous location, if the call to `packmol` failed, then `os.chdir(wd)` would never be executed. By adding it to a finally clause, we make sure we get back home. I also moved the `wd = os.getcwd()` so that it is always in scope.